Fix CEP batching FCFS mode to group CEPs per namespace.

dlapcevic opened the pull request:
Fix CEP (CiliumEndpoint) batching FCFS (First Come First Served) mode to group CEPs per namespace. Initial change was introduced to always add new CEPs to the largest CES (CiliumEndpointSlice), but didn’t take into account splitting them by namespaces, which is a bug. https://github.com/cilium/cilium/pull/20969

Signed-off-by: Dorde Lapcevic <[dordel@google.com](mailto:dordel@google.com)>

alan-kut commented on operator/pkg/ciliumendpointslice/manager.go:
```diff
@@ -379,7 +379,7 @@ func (c *cesMgr) getLargestAvailableCES() *cesTracker {
 		ces.backendMutex.RUnlock()
 
-		if cepCount < c.maxCEPsInCES && cepCount > largestCEPCount {
+		if ces.ces.Namespace == ns && cepCount < c.maxCEPsInCES && cepCount > largestCEPCount {
```
I'd move namespace condition as a last one - it's quicker to compare integers than strings.

dlapcevic replied:
Good suggestion. Done.

dlapcevic committed:
Fix CEP batching FCFS mode to group CEPs per namespace.

Fix CEP (CiliumEndpoint) batching FCFS (First Come First Served) mode to group CEPs per namespace.
Initial change was introduced to always add new CEPs to the largest CES (CiliumEndpointSlice), but didn’t take into account splitting them by namespaces, which is a bug.
https://github.com/cilium/cilium/pull/20969

Signed-off-by: Dorde Lapcevic <dordel@google.com>
